Sources/InteroperabilityTests/TestProvider.swift

```diff
@@ -0,0 +1,179 @@
+  /// `INVALID_ARGUMENT`.
+  public func unaryCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_TestService.Method.UnaryCall.Input>
+  ) async throws -> GRPCCore.ServerResponse.Single<Grpc_Testing_TestService.Method.UnaryCall.Output>
+  {
+    if request.message.responseStatus.isInitialized {
+      let code = Status.Code(rawValue: Int(request.message.responseStatus.code))
+      let status = Status.init(
+        code: code ?? .unknown,
+        message: request.message.responseStatus.message
+      )
+      if let error = GRPCCore.RPCError(status: status) {
+        throw error
+      }
+    }
+
+    if case .UNRECOGNIZED = request.message.responseType {
+      throw RPCError(code: .invalidArgument, message: "The response type is not recognized.")
+    }
+
+    let responseMessage = Grpc_Testing_TestService.Method.UnaryCall.Output.with { response in
+      response.payload = Grpc_Testing_Payload.with { payload in
+        payload.body = Data(repeating: 0, count: Int(request.message.responseSize))
+        payload.type = request.message.responseType
+      }
+    }
+
+    return ServerResponse.Single(message: responseMessage)
+  }
+
+  /// Server gets the default `SimpleRequest` proto as the request. The content of the request is
+  /// ignored. It returns the `SimpleResponse` proto with the payload set to current timestamp.
+  /// The timestamp is an integer representing current time with nanosecond resolution. This
+  /// integer is formated as ASCII decimal in the response. The format is not really important as
+  /// long as the response payload is different for each request. In addition it adds cache control
+  /// headers such that the response can be cached by proxies in the response path. Server should
+  /// be behind a caching proxy for this test to pass. Currently we set the max-age to 60 seconds.
+  public func cacheableUnaryCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_TestService.Method.CacheableUnaryCall.Input>
+  ) async throws
+    -> GRPCCore.ServerResponse.Single<Grpc_Testing_TestService.Method.CacheableUnaryCall.Output>
+  {
+    throw RPCError(code: .unimplemented, message: "The RPC is not implemented.")
+  }
+
+  /// Server implements `streamingOutputCall` by replying, in order, with one
+  /// `StreamingOutputCallResponse` for each `ResponseParameter`s in `StreamingOutputCallRequest`.
+  /// Each `StreamingOutputCallResponse` should have a payload body of size `ResponseParameter.size`
+  /// bytes, as specified by its respective `ResponseParameter`. After sending all responses, it
+  /// closes with OK.
+  public func streamingOutputCall(
+    request: GRPCCore.ServerRequest.Single<
+      Grpc_Testing_TestService.Method.StreamingOutputCall.Input
+    >
+  ) async throws
+    -> GRPCCore.ServerResponse.Stream<Grpc_Testing_TestService.Method.StreamingOutputCall.Output>
+  {
+    return ServerResponse.Stream { writer in
+      for responseParameter in request.message.responseParameters {
```

Comment:
What about the `interval` parameter?

Reply:
I didn't know how to configure the server responses to be send after an interval of time, could you give me a hint?

Reply:
`try await Task.sleep(nanoseconds:)` https://developer.apple.com/documentation/swift/task#suspending-execution

Reply:
Thank you!